src/Microsoft.AspNetCore.Routing/EndpointRoutingMiddleware.cs

```diff
@@ -77,6 +77,18 @@ public async Task Invoke(HttpContext httpContext)
             await _next(httpContext);
         }
 
+        private static void SetEndpointFeature(HttpContext httpContext, EndpointFeature feature)
+        {
+            // An IRouteValuesFeature might have already been set
+            // Copy its RouteValues collection if present
+            var currentRouteValuesFeature = httpContext.Features.Get<IRouteValuesFeature>();
+            feature.RouteValues = currentRouteValuesFeature?.RouteValues;
```

Comment:
Is there a scenario you had in mind for this? Since we're trouncing an existing endpoint feature, I think it's better that we also trounce the route values.

Reply:
It is unlikely anyone would have registered an `IEndpointFeature`. Meanwhile `IRouteValuesFeature` could have been modified in V3 through `HttpRequest.RouteValues`.

Reply:
Right, but who would do that and what would their expectations be? I think it's really wierd to go our of our way to address an unknown (at least to me) scenario.

Reply:
I don't know. It just feels like being a good citizen of the eco-system that if we are replacing the IRouteValuesFeature then we should preserve the underlying values, and I don't see a downside.

Reply:
The downside is that we've shipped a behavior that we have to preserve without knowing why it's valuable -- instead of doing the simplest thing. 

I have a counter proposal that I think you'll like. I'll send a PR